### PR TITLE
Added empty msgid check, improved documentation

### PR DESCRIPTION
**Pull request by isld**
I have added a filter which removes any messages with empty msgid, as they break the header detection. Also, added some documentation with an example for options.header as it was missing from README.md.

### CODE REVIEW

File: tasks/xgettext.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ module.exports = function(grunt) {
             contents += "\n";
         }
 
+        translations = _.filter(translations, function(definition) {
+            return (definition.singular != "");
+        });
```
Comment (arendjr) [inline tasks/xgettext.js]: Maybe you want to use `_.filter(translations, 'singular')` instead. That will remove all falsy values for "singular".

**Comment by arendjr** [comment]
Thanks! I added one little comment, but should be good to merge.

**Comment by arendjr** [comment]
PS.: I don't use this project myself anymore and haven't done any maintenance in quite a while... If you're actively using it and are interested in taking over maintenance, I'll be happy to transfer it :)

**Comment by isld** [comment]
Thank you for the remark, I'm not that familiar with underscore and I didn't even find such a feature documented :)

As for transferring the project - I don't mind, I'm using it actively and I am just about to add a check for methods and variables passed to gettext() because that's the reason I got an empty message in the first place.

**Comment by arendjr** [comment]
Great, thanks! I will transfer to you now :)

**Comment by arendjr** [comment]
Hm, it appears I cannot transfer because, as it says, "isld already has a repository in the grunt-xgettext network". It appears you would have to remove your fork before I can transfer ownership, otherwise I would overwrite your fork, which GitHub prohibits :)

Are you comfortable removing your fork so I can do the transfer?

**Comment by isld** [comment]
I just deleted it, you should be good to go now :)

**Comment by arendjr** [comment]
Done, cheers!

**Pull request merged**